src/backend/distributed/executor/multi_real_time_executor.c

```diff
@@ -235,7 +246,13 @@ MultiRealTimeExecute(Job *job)
 	 * user cancellation request, we can now safely emit an error message (all
 	 * client-side resources have been cleared).
 	 */
-	if (taskFailed)
+	if (sizeLimitIsExceeded)
+	{
+		ereport(ERROR, (errmsg("the max intermediate result size is exceeded, "
+							   "consider to set citus.max_intermediate_result_size to "
```

Comment:
We can probably share a bit more information.

```
ERROR: the intermediate result size exceeded citus.max_intermediate_result_size (currently 1048576KB)
DETAIL: Citus restricts the size of intermediate results of complex subqueries and CTEs to avoid accidentally pulling large result sets into once place.
HINT: To run the current query, set citus.max_intermediate_result_size to a higher value or -1 to disable.
```